akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -90,12 +107,28 @@ object RejectionHandler {
       new BuiltRejectionHandler(cases.result(), notFound, isDefault)
   }
 
-  private sealed abstract class Handler
-  private final case class CaseHandler(pf: PartialFunction[Rejection, Route]) extends Handler
+  private sealed abstract class Handler {
+    def mapResponse(map: HttpResponse ⇒ HttpResponse)(implicit ec: ExecutionContext): Handler
+  }
+  private final case class CaseHandler(pf: PartialFunction[Rejection, Route]) extends Handler {
+    override def mapResponse(map: HttpResponse ⇒ HttpResponse)(implicit ec: ExecutionContext): CaseHandler = {
+      copy(pf.andThen(r ⇒ r.andThen(f ⇒ f.collect {
```

Comment:
With `mapResponse`, I meant `BasicDirectives.mapResponse`, I guess that might not have been clear :)